pkg/internal/exec/proclang_linux.go

```diff
@@ -22,6 +23,14 @@ func FindProcLanguage(pid int32, elfF *elf.File) svc.InstrumentableType {
 		}
 	}
 
+	bytes, err := os.ReadFile(fmt.Sprintf("/proc/%d/environ", pid))
+	if err == nil {
+		t := instrumentableFromEnviron(string(bytes))
+		if t != svc.InstrumentableGeneric {
+			return t
+		}
+	}
+
```

Comment:
I'd first run the code below and, in case the code below returns Generic, I'd run your addition as a last-step differentiation.

The reason is that Beyla would always return .NET if this text is visible in the host-level environment variables (for example, running Beyla in host mode with multiple services as OS processes).

Reply:
after ` findLanguageFromElf`?

Reply:
Yeah, thanks!